main.js

```diff
@@ -1,56 +1,302 @@
+                    document.getElementById('print');
+                    printParent.appendChild(userPhone);
+
+    var userEmail = document.createElement('div');
+                        userEmail.textContent = document.userform.email.value;
+                        document.getElementById('print');
+                        printParent.appendChild(userEmail);
+
+
 }
 
+var userform = document.querySelector("form")
+        userform.addEventListener('submit', check);
+
+
+
+
+function validateFirstName() 
+{
+    var first = document.getElementById('firstname').value;
+   
+        if (!first.match(/^[a-zA-Z]+$/)) 
+
+    {
+        alert('Only letters are allowed for first name.');
+        return false;
+    }
+    
+    return true;
+}
+
+function validateLastName() 
+{
+    var last = document.getElementById('lastname').value;
+   
+        if (!last.match(/^[a-zA-Z]+$/)) 
+
+    {
+        alert('Only letters are allowed for last name.');
+        return false;
+    }
+    
+    return true;
+}
+
+
+
+function validateAddress() {
+
+    var address = document.userform.address.value;
+     
+    if(address !== null && address !== "")
+    {
+
+        return true;
+    }
+    else{
+        alert("Address is required");
+        return false;
+    }
+
+
+}
+
+function validateCity() {
+
+    var city = document.userform.city.value;
+     
+    if(city!== null && city !== "")
+    {
+
+        return true;
+    }
+    else{
+        alert("City is required");
+        return false;
+    }
+
+
+}
+
+function validateState() {
+
+    var state = document.userform.state.value;
+    var lettersOnly = document.userform.state.value;
+
+    if(state.length === 2 && lettersOnly.match(/^[a-zA-Z]+$/))
+
+    {
+
+        return true;
+    }
+
+        else{
+        alert("2 letters required for state");
+        return false;
+    }
+}
+
+
+
+function validateZipcode() {
+    var zipcode = document.userform.zipcode.value;
+    var numbersOnly = document.userform.zipcode.value;
+
+    if(zipcode.length === 5 && numbersOnly.match(/^\d+/)) 
+    {
+            return true; 
+}
+        else { 
+        alert("5 numeric characters required for zipcode");
+        return false;
+
+    }
+
+}
+
+function validatePhoneNumber() {
+    var phonenum= /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
+    if(phone.value.match(phonenum)) {
+      return true;
+    }
+    else {
+      alert("10 numeric characters required for phone number");
+      return false;
+    }
+  }
+
+
 function validateEmail() {
-    var emailID = document.userform.EMail.value;
-    atpos = emailID.indexOf("@");
-    dotpos = emailID.lastIndexOf(".");
+    var emailID = document.userform.email.value;
+     var atpos = emailID.indexOf("@");
+    var dotpos = emailID.lastIndexOf(".");
 
     if (atpos < 1 || (dotpos - atpos < 2)) {
         alert("Please enter correct email ID")
-        document.userform.EMail.focus();
+        document.userform.email.focus();
```

Comment:
It's a bit strange to only offer this auto-focus behavior for one form field. This could be a nice touch for the UI, but for now I'd recommend just removing this to keep it consistent with the rest of the form

Reply:
removed auto focus behavior

Reply:
this hasn't been removed